### PR TITLE
Generated Negative Test Cases for Matter Operational Certificates (NOC, ICAC, RCAC)

### PR DESCRIPTION
**Pull request by emargolis**
Added python script that generates negative test vectors using chip-cert tool. Those new test vectors are used to verify failure scenarios for the following use cases:
  - Conversion of DER encoded operation certificate to CHIP TLV format
  - Conversion of CHIP TLV encoded certificate to DER format
  - Loading and parsing of the CHIP TLV encoded certificate
  - Checking validity of the certificate subject field

As a result of these new test one bug was found and fixed in the ChipDN::DecodeFromTLV() function, where chipAttr should be used instead of attrOID when NodeId/FabricId validity is checked.

Some extra validity checks were added to the following methods:
  - ConvertChipCertToX509Cert()
  - ConvertX509CertToChipCert()

The chip-cert tool was updated and enhanced with more error cases.

### CODE REVIEW

File: credentials/test/operational-certificates-error-cases/Chip-Test-ICAC-Ext-AKID-Length-Invalid-Key.chip [hunk 1]
```diff
@@ -0,0 +1 @@
+N�AWS�ψj*r��S�IA�s�� ���F�k�j�{@�E�"pN� �O�^+��2(DU�V��dA�����6��:��H#aM��SvcJ�a޽�QIj
```
Comment (andy31415) [inline credentials/test/operational-certificates-error-cases/Chip-Test-ICAC-Ext-AKID-Length-Invalid-Key.chip:1]: Github shows a diff here, so I am guessing it thinks this is text.

Should we add `.gitattributes` to treat `*.chip binary` ?

**Review by andy31415 — APPROVED** [review]

File: credentials/test/operational-certificates-error-cases/Chip-Test-ICAC-Ext-AKID-Length-Invalid-Key.chip (hunk 1 above)
Comment (emargolis) [reply credentials/test/operational-certificates-error-cases/Chip-Test-ICAC-Ext-AKID-Length-Invalid-Key.chip:1]: added

**Comment by andy31415** [comment]
@emargolis ASAN claims a memory leak:

```
Direct leak of 16536 byte(s) in 39 object(s) allocated from:
    #0 0x557a8015851e in malloc ../staging/llvm_build/tools/clang/stage2-bins/runtimes/runtimes-x86_64-unknown-linux-gnu-bins/compiler-rt/lib/asan/asan_malloc_linux.cpp:69:3
    #1 0x557a802a25d0 in chip::Platform::MemoryAlloc(unsigned long) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/lib/support/CHIPMem-Malloc.cpp:91:12
    #2 0x557a8022c72f in chip::Credentials::ChipCertificateSet::Init(unsigned char) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/CHIPCert.cpp:81:54
    #3 0x557a801e2567 in TestChipCert_ChipCertLoad_ErrorCases(_nlTestSuite*, void*) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/tests/TestChipCert.cpp:214:23
    #4 0x557a802a5aba in nlTestRunner /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../third_party/nlunit-test/repo/src/nlunit-test.c:213:9
    #5 0x557a801e09c2 in TestChipCert() /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/tests/TestChipCert.cpp:2161:5
    #6 0x557a8018b95a in main /__w/connectedhomeip/connectedhomeip/out/sanitizers/gen/TestChipCert.driver.cpp:32:20
    #7 0x7f50be2c2082 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24082)

Direct leak of 4664 byte(s) in 11 object(s) allocated from:
    #0 0x557a8015851e in malloc ../staging/llvm_build/tools/clang/stage2-bins/runtimes/runtimes-x86_64-unknown-linux-gnu-bins/compiler-rt/lib/asan/asan_malloc_linux.cpp:69:3
    #1 0x557a802a25d0 in chip::Platform::MemoryAlloc(unsigned long) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/lib/support/CHIPMem-Malloc.cpp:91:12
    #2 0x557a8022c72f in chip::Credentials::ChipCertificateSet::Init(unsigned char) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/CHIPCert.cpp:81:54
    #3 0x557a801e3180 in TestChipCert_GetCertType_ErrorCases(_nlTestSuite*, void*) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/tests/TestChipCert.cpp:244:23
    #4 0x557a802a5aba in nlTestRunner /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../third_party/nlunit-test/repo/src/nlunit-test.c:213:9
    #5 0x557a801e09c2 in TestChipCert() /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/tests/TestChipCert.cpp:2161:5
    #6 0x557a8018b95a in main /__w/connectedhomeip/connectedhomeip/out/sanitizers/gen/TestChipCert.driver.cpp:32:20
    #7 0x7f50be2c2082 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24082)
```

**Comment by emargolis** [comment]
> @emargolis ASAN claims a memory leak:
> 
> ```
> Direct leak of 16536 byte(s) in 39 object(s) allocated from:
>     #0 0x557a8015851e in malloc ../staging/llvm_build/tools/clang/stage2-bins/runtimes/runtimes-x86_64-unknown-linux-gnu-bins/compiler-rt/lib/asan/asan_malloc_linux.cpp:69:3
>     #1 0x557a802a25d0 in chip::Platform::MemoryAlloc(unsigned long) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/lib/support/CHIPMem-Malloc.cpp:91:12
>     #2 0x557a8022c72f in chip::Credentials::ChipCertificateSet::Init(unsigned char) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/CHIPCert.cpp:81:54
>     #3 0x557a801e2567 in TestChipCert_ChipCertLoad_ErrorCases(_nlTestSuite*, void*) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/tests/TestChipCert.cpp:214:23
>     #4 0x557a802a5aba in nlTestRunner /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../third_party/nlunit-test/repo/src/nlunit-test.c:213:9
>     #5 0x557a801e09c2 in TestChipCert() /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/tests/TestChipCert.cpp:2161:5
>     #6 0x557a8018b95a in main /__w/connectedhomeip/connectedhomeip/out/sanitizers/gen/TestChipCert.driver.cpp:32:20
>     #7 0x7f50be2c2082 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24082)
> 
> Direct leak of 4664 byte(s) in 11 object(s) allocated from:
>     #0 0x557a8015851e in malloc ../staging/llvm_build/tools/clang/stage2-bins/runtimes/runtimes-x86_64-unknown-linux-gnu-bins/compiler-rt/lib/asan/asan_malloc_linux.cpp:69:3
>     #1 0x557a802a25d0 in chip::Platform::MemoryAlloc(unsigned long) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/lib/support/CHIPMem-Malloc.cpp:91:12
>     #2 0x557a8022c72f in chip::Credentials::ChipCertificateSet::Init(unsigned char) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/CHIPCert.cpp:81:54
>     #3 0x557a801e3180 in TestChipCert_GetCertType_ErrorCases(_nlTestSuite*, void*) /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/tests/TestChipCert.cpp:244:23
>     #4 0x557a802a5aba in nlTestRunner /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../third_party/nlunit-test/repo/src/nlunit-test.c:213:9
>     #5 0x557a801e09c2 in TestChipCert() /__w/connectedhomeip/connectedhomeip/out/sanitizers/../../src/credentials/tests/TestChipCert.cpp:2161:5
>     #6 0x557a8018b95a in main /__w/connectedhomeip/connectedhomeip/out/sanitizers/gen/TestChipCert.driver.cpp:32:20
>     #7 0x7f50be2c2082 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24082)
> ```

Should be fixed with the latest commit

**Review by woody-apple — APPROVED** [review]

**Pull request merged**